src/std/misc/string.ss

```diff
@@ -215,3 +218,37 @@
 	(string-set! str i (random-word-char)))
       str)
     ""))
+
+;; str converts all of its arguments into a single string.
+;; When called without an argument an empty string is returned.
+;;
+;; Examples:
+;;  (str 2.0)               => "2.0"
+;;  (str "hello" ", world") => "hello, world"
+(def* str
+  ((v) (if (string? v) v
+           (format (str-format v) v)))
+  (xs (call-with-output-string
+       (lambda (port)
+         (let loop ((rest xs))
+           (match rest
+             ([v . rest]
+              (if (string? v)
+                (write-string v port)
+                (fprintf port (str-format v) v))
+              (loop rest))
+             (else (void))))))))
+
+;; str-format takes any value and returns a formatting string, which can be
+;; used by the :std/format family of procedures. Considers the :pr method
+;; from :std/misc/repr.
+;;
+;; Examples:
+;;  (str-format 5.0)   => "~f"
+;;  (str-format [1 2]) => "~r"
+(def (str-format v)
+  (def (obj-pr? v) (method-ref v ':pr))
+  (match v
+    ((? (and number? inexact?) v)  "~f")
```

Comment:
```suggestion
    ((? (and number? inexact?))  "~f")
```

Reply:
we don't use `v` so no need to bind it!

Reply:
actually it might more sense to just use cond and the inline form of `?`, no need to match at all.